slosh_lib/src/shell_builtins.rs

```diff
@@ -151,45 +150,83 @@ fn sh_str(vm: &mut SloshVm, registers: &[Value]) -> VMResult<Value> {
     Ok(vm.alloc_string(val))
 }
 
-fn env_var(vm: &mut SloshVm, registers: &[Value]) -> VMResult<Value> {
-    if let (Some(name), None) = (registers.first(), registers.get(1)) {
-        let name = match name {
-            Value::String(h) => vm.get_string(*h),
-            Value::StringConst(i) => vm.get_interned(*i),
-            Value::Symbol(i) => vm.get_interned(*i),
-            _ => {
-                return Err(VMError::new_compile(
-                    "env: requires string or symbol argument",
-                ))
-            }
-        };
-        match env::var(name) {
-            Ok(val) => Ok(vm.alloc_string(val)),
-            Err(VarError::NotPresent) => Ok(vm.alloc_string("".to_string())),
-            Err(err) => Err(VMError::new_compile(format!(
-                "env: error finding env var {name}: {err}"
-            ))),
-        }
-    } else {
-        Err(VMError::new_compile(
-            "$sh: wrong number of args, expected one",
-        ))
+/// Usage: (unset-env "NAME_OF_ENVIRONMENT_VARIABLE")
+///
+/// Takes a string, checks the environment to see if the string is a valid
+/// environment variable and if it is it unsets it.
+///
+/// Section: core
+///
+/// Example:
+/// #t
+#[sl_sh_fn(fn_name = "unset-env")]
+fn unset_env(var: LooseString) {
+    unsafe {
+        env::remove_var(var.as_ref());
     }
 }
 
+/// Usage: (set-env "NAME_OF_ENVIRONMENT_VARIABLE" "Value variable should be assigned")
+///
+/// Takes two strings the first is the name of an environment variable, and the second is the
+/// value it should bind.
+///
+/// Section: core
+///
+/// Example:
+/// #t
+#[sl_sh_fn(fn_name = "set-env")]
+fn set_env(var: LooseString, value: LooseString) {
+    unsafe {
+        env::set_var(var.as_ref(), value.as_ref());
```

Comment:
You need to scrub the input to set_var.  See the Rust docs and example in shell/builtins.rs (export and set_var functions), if you don't this can panic.